core/commands/add.go

```diff
@@ -29,6 +31,7 @@ const progressReaderIncrement = 1024 * 256
 const (
 	progressOptionName = "progress"
 	wrapOptionName     = "wrap-with-directory"
+	hiddenOptionName   = "hidden"
```

Comment:
@gatesvp and @wking `git init` creates the .git/info/exclude file that contains:

```
$ cat .git/info/exclude 
# git ls-files --others --exclude-from=.git/info/exclude
# Lines that start with '#' are comments.
# For a project mostly in C, the following would be a good set of
# exclude patterns (uncomment them if you want to use them):
# *.[oa]
# *~
```

Git uses both the .git/info/exclude and the .gitignore files. The .gitignore files are usually tracked in Git and are not created by `git init`. I think ipfs should do something like that, maybe a `.ipfs/ignore` that is never tracked and some `.ipfsignore` that are usually tracked. So what I suggest is that `ipfs init` creates the `.ipfs/ignore` file with some comments and maybe some default uncommented lines that could be platform specific.

About what Git does on Windows, please note that Git on Windows is not made from exactly the same code base as Git for other platforms. There is a specific Git For Windows project that adds some Windows specific code and I don't know much about this code so I might be wrong about it.

Anyway some people will want hidden files tracked others will not, so we need a way to switch that on and off, and it seems to me that the `.ipfs/ignore` and `.ipfsignore` files are the right place for that.